javascript/javascript2/week2/JS2HW2.js

```diff
@@ -0,0 +1,99 @@
+// console.log(evaluationResult)
+
+//Step3. Now we want to perform the job on all of the titles 
+// splitTitles.forEach(oneSplitedTitle => {
+//     let hasDoubledWord = oneSplitedTitle.map((word, index1) => oneSplitedTitle.indexOf(word) != index1 && index1);
+//     let evaluationResult = hasDoubledWord.filter(hasDoubledWord => hasDoubledWord == false ? false : true)
+//     console.log(evaluationResult)
+// })
+
+
+//Step4. Please note that in the last step we just did some work in according to each element of splitTitles. We have not saved the data in a new array.
+//Then lets save the results in an array. you may think that by just assigning the results of forEach into a new variable you are done.
+//Then lets try it.
+// let specialTitles = splitTitles.forEach(oneSplitedTitle => {
+//     let hasDoubledWord = oneSplitedTitle.map((word, index1) => oneSplitedTitle.indexOf(word) != index1 && index1);
+//     let evaluationResult = hasDoubledWord.filter(hasDoubledWord => hasDoubledWord == false ? false : true)
+//     // console.log(evaluationResult)
+// })
+// console.log(specialTitles)
+
+//Step5. But You see that SpecialTitle is undefined because forEach returns nothing. Then we should use map
+// let specialTitles = splitTitles.map(oneSplitedTitle => {
+//     let hasDoubledWord = oneSplitedTitle.map((word, index1) => oneSplitedTitle.indexOf(word) != index1 && index1);
+//     let evaluationResult = hasDoubledWord.filter(hasDoubledWord => hasDoubledWord == false ? false : true)
+//     // console.log(evaluationResult)
+// })
+// console.log(specialTitles)
+
+//step6. But again line 60 returns undefined. Because in the map we have not returned anything. then
+// let specialTitles = splitTitles.map(oneSplitedTitle => {
+//     let hasDoubledWord = oneSplitedTitle.map((word, index1) => oneSplitedTitle.indexOf(word) != index1 && index1);
+//     let evaluationResult = hasDoubledWord.filter(hasDoubledWord => hasDoubledWord == false ? false : true)
+//     return (evaluationResult)
+// })
+// console.log(specialTitles)
+
+// step7. Now lets return index of the film instead of just a random number
+// let specialTitles = splitTitles.map((oneSplitedTitle, index) => {
+//     let hasDoubledWord = oneSplitedTitle.map((word, index1) => oneSplitedTitle.indexOf(word) != index1 && index1);
+//     let evaluationResult = hasDoubledWord.filter(hasDoubledWord => hasDoubledWord == false ? false : true)
+//     if (evaluationResult.length == 0) {
+//         return (false)
+//     } else {
+//         return (index)
+//     }
+// })
+// console.log(specialTitles)
+
+//Step8. This is the easiest step. We just need to filter the array of the last step.
+let specialTitles = splitTitles.map((oneSplitedTitle, index) => {
+    let hasDoubledWord = oneSplitedTitle.map((word, index1) => oneSplitedTitle.indexOf(word) != index1 && index1);
+    let evaluationResult = hasDoubledWord.filter(hasDoubledWord => hasDoubledWord == false ? false : true)
```

Comment:
Make sure to name your variables better. It's very unclear what this code means; `hasDoubledWord` is an array but sounds like a boolean. Also, why name the index in `map` `index1` instead of just `index`? 

The following code does the same thing as the one on line 85
```
let evaluationResult = hasDoubledWord.filter(word => word) 
```

What should this mapping do, why `&& index1`? 🤔 
```
let hasDoubledWord = oneSplitedTitle.map((word, index1) => oneSplitedTitle.indexOf(word) != index1 && index1); 
```

Reply:
Thank you for your feedback. This is if condition I've used here. if ((oneSplitedTitle.indexOf(word)) !== index){
            return index
        }.

Reply:
Thank you for your feedback. && index1 means if ((oneSplitedTitle.indexOf(word)) !== index){
            return index;
        } and I just wanted to write it summerise.